### ISSUE TITLE
Question on SOC prediction results

### ISSUE THREAD

**Issue by sautee** [KeiLongW/battery-state-estimation#9]
Hello @KeiLongW,

I'm going thru the result models you've put up (thanks for those!). I notice that for the experiments - lstm_soc_percentage_lg_result and lstm_soc_capacity_lg_result.

The predictions on the training dataset are accurate, however the SOC predictions on the test dataset differ widely from actual SOC. I notice this when I run training on the corresponding experiments as well.

Any ideas why that could be? I've attached screenshots of the predictions I'm seeing.

![SOC Capacity Prediction](https://user-images.githubusercontent.com/39371787/201657522-c3161419-fdba-49ae-8281-5855e46542a5.png)
![SOC Percentage Prediction](https://user-images.githubusercontent.com/39371787/201651862-93b136af-2487-4a4f-9d12-7dc5018f7e88.png)

**Comment by sautee** [comment]
To note, I get much better results after running lstm_soc_percentage_lg_all_temp. Here's a screenshot of testing the model against some of the 0degC 'Mixed' datasets.

I'm certain the improved accuracy is the result of training with much more data.

![SOC Percentage - 0degC - All Temps](https://user-images.githubusercontent.com/39371787/201661215-4cabf79a-e457-4f83-b1c4-4641162a52f9.png)

**Comment by KeiLongW** [comment]
Yes, you are correct. Those were some earlier experiments we tested and found that training with limited data will result in an unsatisfactory outcome. Actually, the LG data results presented in the paper were from the mixed cycle test (positive_temp_XXX_steps_mixed_test_result).

**Comment by sautee** [comment]
Thanks much @KeiLongW. I'll run those experiments too.

One more question on this thread. I notice that you've used the activation function 'selu' and loss function 'huber' in all experiments. From literature, I'm supposing it is because these functions are more robust vs other activation/loss functions. But could you share why those two were selected?

**Comment by sautee** [comment]
After reading Self-Normalizing Neural Networks by Günter Klambauer et al. it is clear that “SELUs” induce self-normalizing properties like variance stabilization which in turn avoids exploding and vanishing gradients. Vanishing gradients is one of the issues LSTM's solves. So it makes sense to use SELU as the activation function.

Huber implements both the MSE and MAE loss functions depending on the prediction vs actual delta, which I suppose is useful for this SOC prediction use case.
